src/qilib/data_set/data_set.py

```diff
@@ -214,15 +219,19 @@ def default_array_name(self, default_array_name: str) -> None:
         self._add_metadata_to_storage('default_array_name', default_array_name)
 
     def _verify_set_points(self, set_arrays: DataArrays) -> None:
-        if self._set_arrays != set_arrays:
+        if list(self._set_arrays.values()) != set_arrays:
```

Comment:
`set_arrays` is type `DataArrays` so it can be a list of `DataArrays`, tuple of `DataArrays` or just a `DataArray` but the check only checks for lists so this fails:
```

        x_points = np.array(range(0, 2))
        y_points = np.array(range(0, 2))
        x = DataArray(name="x", label="x-axis", unit="mV", is_setpoint=True, preset_data=x_points)
        y = DataArray(name="y", label="y-axis", unit="mV", is_setpoint=True,
                      preset_data=np.tile(np.array(y_points), [x.size, 1]))
        z = DataArray(name="z", label="z-axis", unit="ma", set_arrays=(x, y),
                      preset_data=np.NaN * np.ones((x_points.size, y_points.size)))
        zz = DataArray(name="zz", label="z-axis", unit="ma", set_arrays=(x, y),
                      preset_data=np.NaN * np.ones((x_points.size, y_points.size)))
        data_set = DataSet(data_arrays=z)
        data_set.add_array(zz)
```

But it should not fail. Also this change did not break any unittest and that is not cool. Can you add a test for this?